### PR TITLE
Tax due controller and FS changes

### PR DESCRIPTION
**Pull request by chandraAndem**

### CODE REVIEW

File: app/views/taxYearOverview.scala.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
 @import views.html.helpers.{breadcrumbHelper, statusHelper}
 @import views.html.templates.main_template
 
-@(taxYear: Int, overview: CalcOverview, transaction: Option[TransactionModel] = None, incomeBreakdown: Boolean = false, deductionBreakdown: Boolean = false)(implicit request: Request[_], messages: Messages, appConfig: config.FrontendAppConfig)
+@(taxYear: Int, overview: CalcOverview, transaction: Option[TransactionModel] = None, incomeBreakdown: Boolean = false, deductionBreakdown: Boolean = false, taxDue: Boolean = false)(implicit request: Request[_], messages: Messages, appConfig: config.FrontendAppConfig)
```
Comment (AlexRimmerHMRC) [inline app/views/taxYearOverview.scala.html:25]: taxDue needs adding to the if statement on line 64 as another or case

File: app/controllers/TaxDueSummaryController.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                                     ) extends BaseController with ImplicitDateFormatter with FeatureSwitching {
+
+  val action: ActionBuilder[MtdItUser] = checkSessionTimeout andThen authenticate andThen retrieveNino andThen retrieveIncomeSources
+
+
+  def showTaxDueSummary(taxYear: Int): Action[AnyContent] = {
+
+    action.async {
+      implicit user =>
+          if (isEnabled(TaxDue)) {
```
Comment (AlexRimmerHMRC) [inline app/controllers/TaxDueSummaryController.scala:57]: Formatting looks off, can you reformat the file?

File: app/controllers/TaxDueSummaryController.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                                        val checkSessionTimeout: SessionTimeoutPredicate,
+                                        val authenticate: AuthenticationPredicate,
+                                        val retrieveNino: NinoPredicate,
+                                        val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                        val calculationService: CalculationService,
+                                        val itvcHeaderCarrierForPartialsConverter: ItvcHeaderCarrierForPartialsConverter,
+                                        val auditingService: AuditingService,
+                                        val financialTransactionsService: FinancialTransactionsService,
+                                        val itvcErrorHandler: ItvcErrorHandler
+                                     ) extends BaseController with ImplicitDateFormatter with FeatureSwitching {
```
Comment (AlexRimmerHMRC) [inline app/controllers/TaxDueSummaryController.scala:48]: The way all of these are injected means that they are all implicit. Reference CalculationController on solving this.
Some of these that are injected don't seem to be used also. Can you check whether some of these are required?
For example, financialTransactionsService is not being used.

File: app/views/taxYearOverview.scala.html (hunk 1 above)
Comment (chandraAndem) [reply app/views/taxYearOverview.scala.html:25]: Added

File: app/controllers/TaxDueSummaryController.scala (hunk 3 above)
Comment (chandraAndem) [reply app/controllers/TaxDueSummaryController.scala]: Made suggested changes.

File: app/controllers/TaxDueSummaryController.scala (hunk 2 above)
Comment (chandraAndem) [reply app/controllers/TaxDueSummaryController.scala]: Reformatted whole file now

**Pull request merged**